### PR TITLE
Manual crafting bug fix

### PR DESCRIPTION
**Pull request by Hackerokuz**
## Summary

Fixed calculation of how many items are required per slot to craft requested amount of items. 

For example:
We need 9 sticks
plank recipe results in 4 sticks
this means 3 planks per slot unless we already have 7 sticks in which case we only need 1 plank per slot

## Information

- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed).
- [ ] This PR is **not** a code change (e.g. documentation, README, examples, ...).

## Checklist

- [x] I have searched the open pull requests for duplicates.
- [x] If code changes were made then they have been tested.

### CODE REVIEW

File: src/main/java/adris/altoclef/tasks/CraftInInventoryTask.java [hunk 1]
```diff
@@ -77,13 +83,15 @@ protected Task onResourceTick(AltoClef mod) {
         // No need to free inventory, output gets picked up.
 
             setDebugState("Crafting in inventory... for " + toGet);
+            if(mod.getModSettings().openInvDuringCrafting() && (client == null || client.currentScreen == null || !client.currentScreen.equals(invScreen))) client.setScreen(invScreen);
```
Comment (codemaker4) [inline src/main/java/adris/altoclef/tasks/CraftInInventoryTask.java:86]: This if-statement looks wrong. If client is null, it will try to open the inventory of the null client? Maybe it should be this: (if the settings allow it, and the client is not null, and (either the current screen doesn't exist, or it is not the inventory screen) then open the inventory)
`mod.getModSettings().openInvDuringCrafting() && client != null && (client.currentScreen == null || !client.currentScreen.equals(invScreen))`

File: src/main/java/adris/altoclef/Settings.java [hunk 2]
```diff
@@ -609,6 +615,11 @@ public int getNetherFastTravelWalkingRange() {
     public BlockPos getHomeBasePosition() {
         return homeBasePosition;
     }
+    
+    
+	public boolean openInvDuringCrafting() {
+		return openInvDuringCrafting;
```
Comment (JamesGreen31) [inline src/main/java/adris/altoclef/Settings.java:621]: This is odd to have in a task not related to crafting. Why is this here?

**Review by JamesGreen31 — CHANGES_REQUESTED** [review]

File: src/main/java/adris/altoclef/Settings.java (hunk 2 above)
Comment (Hackerokuz) [reply src/main/java/adris/altoclef/Settings.java:621]: I was adding a setting to altoclef to be able to toggle an option to allow altoclef to open inventory when crafting in 2x2 grid.

I thought this is the place where settings like this should be added.

Did I misunderstand the usage of this class?

I didn't find any other files that specify settings in altoclef.

This is what the setting makes altoclef do: https://www.youtube.com/watch?v=NUdvg6aFYss
